### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by akos29**
### Bookstore: add reducers and actions

In this request majorly:

- Add Redux Toolkit 
- Add files structure using a "Features folder" approach and ducks pattern for redux files
- Add books actions and default reducer
- Add categories actions and reducer 
- Add configure the Redux Store

### CODE REVIEW

**Review by Gambit142 — COMMENTED** [review]
Hi @akos29 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

In your README.md file, kindly make the following changes:

- Kindly remove the Microverse logo. If you have your logo, you can use it, if not please remove the logo.
- Kindly add the **_Future Features_** section in your _**README.md**_ file.  Under this section, kindly add 1 - 3 features that you or other contributors can work on in the future. 👍 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/components/redux/modules/booksSlice.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    },
+    progressUpdate(state, action) {
+      const book = state.find((book) => book.id === action.payload);
+      book.progress += 1;
+    },
+  },
+});
+
+export const { bookAdded, progressUpdate } = booksSlice.actions;
+export default booksSlice.reducer;
```
Comment (Gopxfs) [inline src/components/redux/modules/booksSlice.js:22]: Great job creating a Redux Slice 😄! As requested in the project requirements: 
image 

Kindly consider that you still need to implement some of Redux's functionalities 😉. Please make sure to: 
- [x] Use [Redux Thunk](https://github.com/reduxjs/redux-thunk) to connect with your API. Currently, your project isn't adding or removing books. Keep in mind that you need to keep the functionality after refactoring your code with Redux's tools. 

- [x] Refactor your _add book_ and _remove book_ to persist changes in the server. This means that you should hit the API to update your book list after adding & removing any book. So, when the user comes back later, the page will display the updated book list 😉. 

> Check more about the Bookstore API [on this link](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739).

**Review by Gopxfs — CHANGES_REQUESTED** [review]
# ♻️ STATUS: REQUIRED CHANGES ♻️

## Hi @akos29 👋,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights :rocket:
✔️ No linter errors.
✔️ Great code structure. Well done!
✔️ Implemented Redux Slice.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 🗒️  
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Gopxfs in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by prolajumokeoni — COMMENTED** [review]
Hi @akos29 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted  a wrong pr for the use Redux in React components milestone

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by emmiiorji — COMMENTED** [review]
Hi @akos29, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a wrong pull request for the `Bookstore: add reducers and actions ` milestone. Kindly note that every milestone require a separate pull request containing the changes required for the milestone.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/components/redux/modules/booksSlice.js (hunk 1 above)
Comment (akos29) [reply src/components/redux/modules/booksSlice.js:22]: Thanks @Gopxfs  updated accordingly

**Review by franclobo — APPROVED** [review]
Hi @akos29 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors ✔️ 
- Correct Git flow ✔️ 
- Descriptive PR ✔️ 
- Professional README file ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Invalid Code Review Request ❌

Hi @akos29, This is @sja-thedude your code reviewer.

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

You have submitted a project, that is already approved and is of different milestones. This project is Bookstore: add reducers and actions and you have asked review for Bookstore: Use Redux in React components. Kindly make sure to submit a new pull request for each milestone.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by Davidosky007 — COMMENTED** [review]
Hi @akos29 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.
invalide

## Invalid Code Review Request

You have submitted a Wrong PR for this milestone `Bookstore: add reducers and actions` kindly open another PR and implement all requirements for this milestone on them and submit for reviewing 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Pull request merged**